Sources/Rendering/OpenGL/ImageResliceMapper/index.js

```diff
@@ -0,0 +1,1450 @@
+      model.renderable.setSlicePlane(plane);
+      resGeomString = resGeomString.concat(
+        `Plane${model.renderable.getSlicePlane()?.getMTime()}`
+      );
+      if (image) {
+        resGeomString = resGeomString.concat(`Image${image.getMTime()}`);
+      }
+    }
+
+    if (!model.resliceGeom || model.resliceGeomUpdateString !== resGeomString) {
+      if (model.renderable.getSlicePolyData()) {
+        model.resliceGeom = model.renderable.getSlicePolyData();
+      } else if (model.renderable.getSlicePlane()) {
+        const bounds = image ? imageBounds : [0, 1, 0, 1, 0, 1];
+        if (!orthoSlicing) {
+          const cube = vtkCubeSource.newInstance();
+          cube.setCenter(
+            0.5 * (bounds[0] + bounds[1]),
+            0.5 * (bounds[2] + bounds[3]),
+            0.5 * (bounds[4] + bounds[5])
+          );
+          cube.setXLength(bounds[1] - bounds[0]);
+          cube.setYLength(bounds[3] - bounds[2]);
+          cube.setZLength(bounds[5] - bounds[4]);
+          const cutter = vtkCutter.newInstance();
+          cutter.setInputConnection(cube.getOutputPort());
+          cutter.setCutFunction(model.renderable.getSlicePlane());
+          const pds = vtkClosedPolyLineToSurfaceFilter.newInstance();
+          pds.setInputConnection(cutter.getOutputPort());
+          pds.update();
+          model.resliceGeom = pds.getOutputData();
+        } else {
+          const ptsArray = new Float32Array(12);
+          const o = model.renderable.getSlicePlane().getOrigin();
+          console.log(o);
+          if (orthoAxis === 0) {
+            ptsArray[0] = o[0];
+            ptsArray[1] = bounds[2];
+            ptsArray[2] = bounds[4];
+            ptsArray[3] = o[0];
+            ptsArray[4] = bounds[3];
+            ptsArray[5] = bounds[4];
+            ptsArray[6] = o[0];
+            ptsArray[7] = bounds[2];
+            ptsArray[8] = bounds[5];
+            ptsArray[9] = o[0];
+            ptsArray[10] = bounds[3];
+            ptsArray[11] = bounds[5];
+          } else if (orthoAxis === 1) {
+            ptsArray[0] = bounds[0];
+            ptsArray[1] = o[1];
+            ptsArray[2] = bounds[4];
+            ptsArray[3] = bounds[1];
+            ptsArray[4] = o[1];
+            ptsArray[5] = bounds[4];
+            ptsArray[6] = bounds[0];
+            ptsArray[7] = o[1];
+            ptsArray[8] = bounds[5];
+            ptsArray[9] = bounds[1];
+            ptsArray[10] = o[1];
+            ptsArray[11] = bounds[5];
+          } else {
+            ptsArray[0] = bounds[0];
+            ptsArray[1] = bounds[2];
+            ptsArray[2] = o[2];
+            ptsArray[3] = bounds[1];
+            ptsArray[4] = bounds[2];
+            ptsArray[5] = o[2];
+            ptsArray[6] = bounds[0];
+            ptsArray[7] = bounds[3];
+            ptsArray[8] = o[2];
+            ptsArray[9] = bounds[1];
+            ptsArray[10] = bounds[3];
+            ptsArray[11] = o[2];
+          }
```

Comment:
Instead of differentiating the three cases, an equivalent way of doing it is:
```
const otherAxes = [(orthoAxis + 1) % 3, (orthoAxis + 2) % 3].sort();
let ptIdx = 0;
for (let i = 0; i < 2; ++i) {
  for (let j = 0; j < 2; ++j) {
    ptsArray[ptIdx + orthoAxis] = o[orthoAxis];
    ptsArray[ptIdx + otherAxes[0]] = bounds[2 * otherAxes[0] + j];
    ptsArray[ptIdx + otherAxes[1]] = bounds[2 * otherAxes[1] + i];
    ptIdx += 3;
  }
}
```